### ISSUE TITLE
Playlist overwritten when adding track in subsonic client

### ISSUE THREAD

**Issue by Merkwurdichliebe** [navidrome/navidrome#2154]
### Description

A `m3u8` playlist imported from the music folder is completely overwritten if a track is added to it from a subsonic client. The result is a playlist containing only the added track.

### Expected Behaviour

The track should be appended to the end of the playlist.

### Steps to reproduce

1. Import the playlist into navidrome.
2. In a subsonic client (play:Sub in my case), play a track and add it to an existing playlist.
3. Refresh the navidrome web UI.
4. The playlist only contains the new track.
5. 
### Platform information

 - Navidrome version: 0.49.1
 - Browser and version: Firefox 109.0.1
 - Operating System: macOS 12.6.2

### Additional information

Any other information that may be relevant or give context to the problem.

 - play:Sub 2022.3 (3358)
 - Couldn't try it with iSub because it has no "add to playlist" function AFAIK.

**Comment by deluan** [comment]
I just tried this with DSub and the track is simply appended to the playlist as expected. Seems to be an issue with play:Sub. 
 Please try with a different client, maybe Substreamer?

**Comment by Merkwurdichliebe** [comment]
I can confirm that it doesn't happen with Substreamer.

Also, the issue occurs only with a freshly imported playlist and the first appended track from play:Sub, which erases it. Further tracks added to the playlist from play:Sub are appended properly. I've deleted the playlist and reimported it twice to make sure the behavior was consistent.

**Comment by deluan** [comment]
Maybe @MichaelBechHansen (play:Sub dev) can help with that? You should also open an issue with [play:Sub's support](http://michaelsapps.dk/playsubapp/playsub%20support.html)

**Comment by dweymouth** [comment]
Having just worked on integrating playlist support in my own new client I have a guess as what this could be. The [`createPlaylist`](http://www.subsonic.org/pages/api.jsp#createPlaylist) Subsonic API can, confusingly, also be used to update a playlist, and the documentation is kind of ambiguous in the update case as to whether the supplied trackIDs should be appended to, or entirely replace, the tracks already in the playlist.

@deluan could you check if Navidrome appends or replaces tracks when `createPlaylist` is called with the `playlistId` and multiple `songId` parameters?

In any case, I think subsonic clients should only use `createPlaylist` to create new playlists, and the much less ambiguous `updatePlaylist` API to make changes to an existing playlist

**Comment by MichaelBechHansen** [comment]
When adding to an existing playlist, play:Sub calls the createPlaylist endpoint with a list of all the trackIds that the resulting playlist should contain.  
On the serverside this should update the playlist accordingly.
Verified to work with latest play:Sub against latest subsonic.org server.

**Comment by deluan** [comment]
@MichaelBechHansen thanks for the reply. So you don't use the `updatePlaylist` at all, right? Can you post here an example of a call to `createPlaylist` generated by play:Sub when it is used to update the playlist?

**Comment by MichaelBechHansen** [comment]
Yep - play:Sub is not using the updatePlaylist endpoint for editing the playlist contents.
It does use the updatePlaylist endpoint when editing playlist name/comment/public-private though.

A call to createPlaylist for updating/editing from play:Sub:
Endpoint URL:  http://SERVER/rest/createPlaylist.view
HTTP Method: POST
HTTP Body: `songId=28357&songId=29042&songId=63559&songId=63555&playlistId=23&name&f=json&c=playSub_iPhone12Pro&t=REDACTED&u=REDACTED&v=1.9.0=000&s=REDACTED`

**Comment by MichaelBechHansen** [comment]
BTW Just to give some background for why using createPlaylist instead of updatePlaylist:

For a client app to make multiple changes to a playlist, reordering, adding, deleting tracks, updating via createPlaylist is as far as I can see the only feasible option.
WRT playlist content, updatePlaylist allows for two things:
- adding song(s) to the end of the playlist (songIdToAdd parameter)
- deleting a song by index into the existing playlist tracks (songIndexToRemove)

This makes some things hard:
- reordering tracks, f.ex. move last track to first position.
- adding songs in other places than the end.
- if/when a playlist is edited by multiple clients, the incremental edits with updatePlaylist can result in playlists that does not match what ANY of the editing clients made.  With createPlaylist, the last writer wins, and gets the result that was edited locally (so at least one editing client gets what they want).

This is - as far as I remember - the primary reasons for going with createPlaylist for editing playlist content instead of updatePlaylist.

**Comment by deluan** [comment]
Based on the discussions above, and confirming that other clients for both iOS (Substreamer) and Android (DSub) works correctly, I'm closing this. Let me know if there are any other concerns.
